rust/hyperlane-base/src/contract_sync/broadcast.rs

```diff
@@ -0,0 +1,52 @@
+use std::sync::Arc;
+
+use derive_new::new;
+use eyre::Result;
+use hyperlane_core::H512;
+use tokio::sync::{
+    mpsc::{Receiver as MpscReceiver, Sender as MpscSender},
+    Mutex,
+};
+
+#[derive(Debug, Clone, new)]
+/// Wrapper around a vec of mpsc senders that broadcasts messages to all of them.
+/// This is a workaround to get an async interface for `send`, so senders are blocked if any of the receiving channels is full,
+/// rather than overwriting old messages (as the `broadcast` channel ring buffer implementation does).
```

Comment:
https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#method.blocking_send is this relevant?